android/src/main/java/com/adyenreactnativesdk/ui/LoadingDialogFragment.kt

```diff
@@ -0,0 +1,39 @@
+package com.adyenreactnativesdk.ui
+
+import android.graphics.Color
+import android.graphics.drawable.ColorDrawable
+import android.os.Bundle
+import android.view.LayoutInflater
+import android.view.View
+import android.view.ViewGroup
+import android.widget.Button
+import androidx.fragment.app.DialogFragment
+import com.adyenreactnativesdk.R
+
+interface Cancelable {
+    fun canceled()
+}
+
+class LoadingDialogFragment(private val cancelable: Cancelable) : DialogFragment() {
+
+    override fun onCreateView(inflater: LayoutInflater, container: ViewGroup?, savedInstanceState: Bundle?): View? {
+        isCancelable = false
+        return inflater.inflate(R.layout.loading, container, false)
+    }
+
+    override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
+        super.onViewCreated(view, savedInstanceState)
+        requireDialog().window?.setBackgroundDrawable(ColorDrawable(Color.TRANSPARENT))
+
+        val button = view.findViewById<Button>(R.id.loading_button)
+        button.setOnClickListener {
+            cancelable.canceled()
+        }
+    }
+
+    companion object {
+        fun newInstance(cancelable: Cancelable): LoadingDialogFragment {
```

Comment:
it's not a good practice to pass a callback in the `newInstance()` function and in the constructor. instead have a class level field as following:
```kotlin
var cancellable: Cancellable? = null
```

then set it to action handler where you instantiate the fragment.

Reply:
Fix here [38727d5](https://github.com/Adyen/adyen-react-native/pull/55/commits/38727d5a0b44c9205b5f9b2e133004a719e04ece)